docs/scripts/reference.py

```diff
@@ -358,8 +370,7 @@ def gen_reference_tagformatters():
         elif _is_tag_formatter_cls(obj):
             tag_formatter_classes[name] = obj
 
-    out_file.parent.mkdir(parents=True, exist_ok=True)
-    with out_file.open("w", encoding="utf-8") as f:
+    with mkdocs_gen_files.open(out_path, "w", encoding="utf-8") as f:
```

Comment:
The fix for the restarting loop was to use `mkdocs_gen_files.open()` instead of `open()` to file writes go through `mkdocs-gen-files`